crates/anvil-polkadot/tests/it/snapshot.rs

```diff
@@ -0,0 +1,598 @@
+    let baltathar_account = Account::from(subxt_signer::eth::dev::baltathar());
+    let baltathar_addr = Address::from(ReviveAddress::new(baltathar_account.address()));
+
+    // Initialize a random account.
+    let transfer_amount = U256::from(16e17);
+    let (dest_addr, _) =
+        node.eth_transfer_to_unitialized_random_account(alith_addr, transfer_amount, None).await;
+
+    // Make another regular transfer between known accounts.
+    let transfer_amount = U256::from(1e17);
+    let transaction =
+        TransactionRequest::default().value(transfer_amount).from(baltathar_addr).to(alith_addr);
+    let _ = node.send_transaction(transaction, None).await.unwrap();
+
+    // Revert to a block before the transactions have been sent.
+    let snapshot_id = U256::from_str_radix(id.trim_start_matches("0x"), 16).unwrap();
+    assert_eq!(snapshot_id, U256::ZERO);
+    let reverted =
+        unwrap_response::<bool>(node.eth_rpc(EthRequest::EvmRevert(snapshot_id)).await.unwrap())
+            .unwrap();
+    assert!(reverted);
+    assert_block_number(&mut node, 5).await;
+    let id = unwrap_response::<String>(node.eth_rpc(EthRequest::EvmSnapshot(())).await.unwrap())
+        .unwrap();
+    assert_eq!(id, "0x1".to_string());
+
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(Some(U256::from(1)), None)).await.unwrap())
+        .unwrap();
+    assert_block_number(&mut node, 6).await;
+
+    let txs_in_block = unwrap_response::<U256>(
+        node.eth_rpc(EthRequest::EthGetTransactionCountByNumber(
+            alloy_eips::BlockNumberOrTag::Latest,
+        ))
+        .await
+        .unwrap(),
+    )
+    .unwrap();
+    assert_eq!(txs_in_block, U256::from(2));
+
+    // Now make two more txs again with same senders, with different nonces than the actual
+    // accounts nonces at block 5.
+    let transfer_amount = U256::from(1e15);
+    let transaction =
+        TransactionRequest::default().value(transfer_amount).from(baltathar_addr).to(alith_addr);
+    let _ = node.send_transaction(transaction, None).await.unwrap();
+    let _ = node
+        .send_transaction(
+            TransactionRequest::default().value(transfer_amount).from(alith_addr).to(dest_addr),
+            None,
+        )
+        .await
+        .unwrap();
+
+    let snapshot_id = U256::from_str_radix(id.trim_start_matches("0x"), 16).unwrap();
+    let reverted =
+        unwrap_response::<bool>(node.eth_rpc(EthRequest::EvmRevert(snapshot_id)).await.unwrap())
+            .unwrap();
+    assert!(reverted);
+    assert_block_number(&mut node, 5).await;
+
+    let txs_in_block = unwrap_response::<U256>(
+        node.eth_rpc(EthRequest::EthGetTransactionCountByNumber(
+            alloy_eips::BlockNumberOrTag::Latest,
+        ))
+        .await
+        .unwrap(),
+    )
+    .unwrap();
+    assert_eq!(txs_in_block, U256::ZERO);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_mine_with_txs_in_mempool_after_revert() {
```

Comment:
AFAICT, this is already being tested in `test_balances_and_txs_index_after_evm_revert`